### PR TITLE
decouple data into leaves and top half MT

### PR DESCRIPTION
**Pull request by schomatis**
Closes https://github.com/filecoin-project/merkle_light/issues/16.

### CODE REVIEW

**Comment by schomatis** [comment]
The code (as used in https://github.com/filecoin-project/rust-fil-proofs/pull/642) seems to be working, need to update the internal tests here for Travis.

**Comment by schomatis** [comment]
Passing the tests, ready for initial round of review, but #15 should be reviewed and merged first.

**Comment by dignifiedquire** [comment]
Can you please make sure the benchmarks still compile, I noticed that they are failing on `master`

**Comment by schomatis** [comment]
@dignifiedquire Could you point me to the failure to better understand what I should fix please? Also, we should focus on #15 first (which currently isn't working) on which this PR is based before reviewing this one (the failure in the bench may be rooted there).

**Comment by dignifiedquire** [comment]
I created a new issue for this here: https://github.com/filecoin-project/merkle_light/issues/18

File: merkle/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "merkletree"
-version = "0.4.2-alpha.0"
```
Comment (dignifiedquire) [inline merkle/Cargo.toml]: BLOCKING: please undo, this change happens automatically when I run the release script

**Comment by schomatis** [comment]
yes, we're due a minor version bump since my last PR..

**Comment by schomatis** [comment]
@dignifiedquire dropped the bump commit

**Pull request merged**